x-pack/legacy/plugins/siem/public/components/formatted_date/index.tsx

```diff
@@ -8,26 +8,19 @@ import moment from 'moment-timezone';
 import * as React from 'react';
 import { FormattedRelative } from '@kbn/i18n/react';
 
-import { useUiSetting$ } from '../../lib/kibana';
-
-import {
-  DEFAULT_DATE_FORMAT,
-  DEFAULT_DATE_FORMAT_TZ,
-  DEFAULT_TIMEZONE_BROWSER,
-} from '../../../common/constants';
+import { useDateFormat, useTimeZone } from '../../hooks';
 import { getOrEmptyTagFromValue } from '../empty_value';
 import { LocalizedDateTooltip } from '../localized_date_tooltip';
 import { getMaybeDate } from './maybe_date';
 
 export const PreferenceFormattedDate = React.memo<{ value: Date }>(({ value }) => {
-  const [dateFormat] = useUiSetting$<string>(DEFAULT_DATE_FORMAT);
-  const [dateFormatTz] = useUiSetting$<string>(DEFAULT_DATE_FORMAT_TZ);
-  const [timezone] = useUiSetting$<string>(DEFAULT_TIMEZONE_BROWSER);
+  const dateFormat = useDateFormat();
+  const timeZone = useTimeZone();
 
   return (
     <>
-      {dateFormat && dateFormatTz && timezone
-        ? moment.tz(value, dateFormatTz === 'Browser' ? timezone : dateFormatTz).format(dateFormat)
+      {dateFormat && timeZone
+        ? moment.tz(value, timeZone).format(dateFormat)
```

Comment:
It looks like `dateFormat` and `timeZone` are both going to be defined at all times so it is possible to clean this up more by dropping the optional checks for them and the unit tests?

It would become:

```ts
export const PreferenceFormattedDate = React.memo<{ value: Date }>(({ value }) => {
  const dateFormat = useDateFormat();
  const timeZone = useTimeZone();
  return <>{moment.tz(value, timeZone).format(dateFormat)}</>;
});
```

That would make it a pretty awesome code read.

Reply:
👍 

FYI`dateFormat` can still be an empty string as described above, but in that case they'll just get the 8601 format: 
<img width="889" alt="Hosts_-_Kibana" src="https://user-images.githubusercontent.com/657252/71856650-79cf9c00-3099-11ea-98b9-1347cbde0162.png">